lib/models.ts

```diff
@@ -0,0 +1,35 @@
+import { ComponentsDefinitionV10X } from './components-types-v1_0_x';
+
+export enum DirectiveType {
+    editable = 'editable',
+    container = 'container',
+    image = 'image',
+    html = 'html',
+    if = 'if',
+    slideshow = 'slideshow',
+    link = 'link',
+    media = 'media',
+    interactive = 'interactive',
+
+    unknown = 'unknown',
+}
+
+export interface ParsedComponentsDefinition {
+    components: {
+        [name: string]: {
+            component: ComponentsDefinitionV10X['components'][0];
```

Comment:
I wonder how this will work out for future versions. At some point we might have a `ComponentsDefinitionV11X`. We might get something like `ComponentsDefinitionV10X['components'][0] | ComponentsDefinitionV11X['components'][0]` here.

Reply:
Besides that, it does not become more readable with types like `ComponentsDefinitionV10X['components'][0]` and `ParsedComponentsDefinition['components']['name']`. That would be something that we could address in `generateComponentTypes.ts`, right?

Reply:
Added generic types. Probably it solves it partly at least.